src/layout/verticalPanel/allApplicationList.ts

```diff
@@ -0,0 +1,194 @@
+                    this.parentalControlsManager.shouldShowApp(appInfo)
+                );
+            })
+            .sort((a, b) =>
+                a.get_display_name().localeCompare(b.get_display_name())
+            );
+
+        for (let appInfo of appsInstalled) {
+            let icon = new St.Icon({
+                icon_size: 32,
+                gicon: appInfo.get_icon(),
+            });
+
+            const entry = new SearchResultEntry(
+                icon,
+                appInfo.get_display_name(),
+                // The remote search provider also provides a description field, but the app search does not
+                appInfo.get_description(),
+                true
+            );
+            entry.connect('primary-action', () => {
+                // It's important that we do this first because it will remove the focus grab that we use.
+                // This has the effect of restoring focus to the actor that was focused when we first grabbed it.
+                // So if we want to focus a newly created window we need to be sure to do it after we close the search view.
+                this.resetAndClose();
+
+                const app = Shell.AppSystem.get_default().lookup_app(
+                    appInfo.get_id()
+                );
+                if (app) {
+                    Me.msWindowManager.openApp(
+                        app,
+                        Me.msWorkspaceManager.getActiveMsWorkspace()
+                    );
+                }
+            });
+            this.addResult(entry);
+        }
+    }
+
+    addResult(resultEntry: SearchResultEntry): void {
+        this.add_child(resultEntry);
+        if (this.resultEntryList.length === 1) {
+            this.selectResult(resultEntry);
+        }
+    }
+
+    selectResult(entry: SearchResultEntry): void {
+        if (this.entrySelected) {
+            this.entrySelected.setSelected(false);
+        }
+        this.entrySelected = entry;
+        this.entrySelected.setSelected(true);
+        this.emit('result-selected-changed', this.entrySelected);
+    }
+
+    selectNext() {
+        if (this.entrySelected == null) {
+            this.entrySelected = this.resultEntryList[0];
+            this.selectResult(this.entrySelected);
+            return;
+        }
+        const currentIndex = this.resultEntryList.indexOf(this.entrySelected);
+        const nextEntry = this.resultEntryList[currentIndex + 1];
+        if (nextEntry) {
+            this.selectResult(nextEntry);
+        }
+    }
+
+    selectPrevious() {
+        if (this.entrySelected == null) {
+            this.entrySelected = this.resultEntryList[0];
+            this.selectResult(this.entrySelected);
+            return;
+        }
+        const currentIndex = this.resultEntryList.indexOf(this.entrySelected);
+        const previousEntry = this.resultEntryList[currentIndex - 1];
+        if (previousEntry) {
+            this.selectResult(previousEntry);
+        }
+    }
```

Comment:
```suggestion
    selectPrevious() {
        const entries = this.resultEntryList;
        const currentIndex = entries.indexOf(this.entrySelected);
        const previousEntry = currentIndex === -1 ?  entries[0] : entries[currentIndex - 1];
        if (previousEntry !== undefined) {
            this.selectResult(previousEntry);
        }
    }
```

Reply:
Here you assume that the list has a length > 0 too. 
I endded up discard the function if there is no length or no current selection

Reply:
No, I do not. If entries[0] is undefined it will fail the check at row 182.